packages/mongodb-chatbot-ui/src/Chatbot.tsx

```diff
@@ -17,6 +18,7 @@ export type ChatbotProps = {
   shouldStream?: boolean;
   tck?: string;
   user?: User;
+  fetchOptions?: ConversationFetchOptions;
```

Comment:
@nlarew react-y question: rather than passing object of options here, wondering if you think there could be a better way of doing this for if the stuff in that object changes. 

like if the user is having this be an object regulated by React state (say if auth token changes), right now i think that'd cause a whole chatbot re-render, which i don't think is what we'd want here.

Reply:
re: memoization, you have to be careful passing objects as props because they can be recreated every render and lose their identity.

For example consider this code:

```
// A. object props defined inline
<Chatbot fetchOptions={{ headers: { "X-My-Api-Key": "abcdefg" } }}>
  {children}
</Chatbot>

// B. static object
// Define customFetchOptions outside of the component
const customFetchOptions = { headers: { "X-My-Api-Key": "abcdefg" } };
// ... later on in the component render
<Chatbot fetchOptions={customFetchOptions}>
  {children}
</Chatbot>
```

Version A will naively cause a full tree rerender every time because the value of `fetchOptions` is a new object (with the same keys & values) every render. We would need to add additional equality checks / memoization inside our component tree.

Version B will share the identity of `customFetchOptions` between rerenders.

Reply:
per convo, let's keep as is w/ option B.